Tasks/task2.html

```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
+
+<head>
+    <title>Task 2</title>
+</head>
+
+<body>
+    <script>
+        const recipes = {
+            recipes: [],
+            addRecipe(name) {
+                this.recipes.push({ id: Math.random(), name, products: [] });
+            },
+            removeRecipe(id) {
+                this.recipes = this.recipes.filter((r) => r.id != id);
+            },
+            addProduct(recipeId, name) {
+                const recipe = this.recipes.find((r) => r.id == recipeId);
+                recipe.products.push({ id: Math.random(), name });
+            },
+            removeProduct(id) {
+                this.recipes.forEach((recipe) => {
+                    recipe.products = recipe.products.filter((p) => p.id != id);
+                });
+            },
+        };
+
+        function render() {
+            document.body.innerHTML = `
+                <h1>CookAsistant</h1>
+                <input placeholder="Recipe Name" />
+                <button onclick="recipes.addRecipe(this.previousElementSibling.value)">Add Recipe</button>
```

Comment:
Its probably static element of page, it could be extracted into html itself.
instead refer to body could change innerHTML inside a div bellow this button.